tsp-output/@azure-tools/typespec-csharp/src/Generated/Internal/ClientPipelineExtensions.cs

```diff
@@ -1,49 +1,36 @@
-// <auto-generated/>
 
-#nullable disable
+#nullable enable
 
-using System;
 using System.ClientModel;
 using System.ClientModel.Primitives;
-using System.Threading;
 using System.Threading.Tasks;
 
 namespace OpenAI
 {
     internal static class ClientPipelineExtensions
     {
-        public static async ValueTask<PipelineResponse> ProcessMessageAsync(this ClientPipeline pipeline, PipelineMessage message, RequestOptions requestContext, CancellationToken cancellationToken = default)
+        public static async ValueTask<PipelineResponse> ProcessMessageAsync(this ClientPipeline pipeline, PipelineMessage message, RequestOptions options)
         {
             await pipeline.SendAsync(message).ConfigureAwait(false);
 
-            if (message.Response == null)
+            if (message.Response!.IsError && options.ErrorOptions == ClientErrorBehaviors.Default)
```

Comment:
We should clarify the positive case where we throw, instead of the negative case.  The method should end returning a successful result to clarify what's happening in it.

Reply:
https://github.com/Azure/autorest.csharp/issues/4485